ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/deneb/BlobSidecar.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright ConsenSys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.datastructures.execution.versions.deneb;
+
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.infrastructure.ssz.containers.Container8;
+import tech.pegasys.teku.infrastructure.ssz.primitive.SszBytes32;
+import tech.pegasys.teku.infrastructure.ssz.primitive.SszUInt64;
+import tech.pegasys.teku.infrastructure.ssz.tree.TreeNode;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.kzg.KZGCommitment;
+import tech.pegasys.teku.kzg.KZGProof;
+import tech.pegasys.teku.spec.datastructures.type.SszKZGCommitment;
+import tech.pegasys.teku.spec.datastructures.type.SszKZGProof;
+
+public class BlobSidecar
+    extends Container8<
+        BlobSidecar,
+        SszBytes32,
+        SszUInt64,
+        SszUInt64,
+        SszBytes32,
+        SszUInt64,
+        Blob,
+        SszKZGCommitment,
+        SszKZGProof> {
+
+  BlobSidecar(final BlobSidecarSchema blobSidecarSchema, final TreeNode backingTreeNode) {
+    super(blobSidecarSchema, backingTreeNode);
+  }
+
+  public BlobSidecar(
+      BlobSidecarSchema schema,
+      Bytes32 blockRoot,
+      UInt64 index,
```

Comment:
Not sure if we should create a [BlobIndex](https://github.com/arnetheduck/eth2.0-specs/blob/back-to-the-decoupled-future/specs/deneb/beacon-chain.md#custom-types) custom type instead of using UInt64, but seems fine to me. Do you have an opinion @tbenr

Reply:
Yep, I hesitated to create a new type but then thought `UInt64` could be fine...

Reply:
we don't  have `Slot`, `Epoch`, `CommitteeIndex`, `ValidatorIndex` and a lot of others. So I think we don't need dedicated class for this one too